src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -535,15 +537,16 @@ private JsonRpcResult ReadResult(JToken json)
         {
             Requires.NotNull(json, nameof(json));
 
-            JToken id = json["id"];
+            RequestId id = this.NormalizeId(json["id"].ToObject<RequestId>());
```

Comment:
Don't we need a null check here just as in ReadRequest?

Reply:
No. Results *always* have id properties.